File: OrchidCore/src/main/java/com/eden/orchid/api/theme/assets/AssetHolder.java
package com.eden.orchid.api.theme.assets;

import java.util.List;

public interface AssetHolder {

    AssetHolder getAssetHolder();

    void addAssets();

    default void withNamespace(String namespace, Runnable cb) {
        getAssetHolder().withNamespace(namespace, cb);
    }

    default JsPage addJs(JsPage jsAsset) {
        return getAssetHolder().addJs(jsAsset);
    }

    default JsPage addJs(String jsAsset) {
        return getAssetHolder().addJs(jsAsset);
    }

    default CssPage addCss(CssPage cssAsset) {
        return getAssetHolder().addCss(cssAsset);
    }

    default CssPage addCss(String cssAsset) {
        return getAssetHolder().addCss(cssAsset);
    }

    default AssetPage addAsset(AssetPage asset) {
        return getAssetHolder().addAsset(asset);
    }

    default AssetPage addAsset(String asset) {
        return getAssetHolder().addAsset(asset);
    }

    default List<JsPage> getScripts() {
        return getAssetHolder().getScripts();
    }

    default List<CssPage> getStyles() {
        return getAssetHolder().getStyles();
    }
    
    default void clearAssets() {
        getAssetHolder().clearAssets();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/theme/assets/AssetHolder.kt
package com.eden.orchid.api.theme.assets

interface AssetHolder {

    fun getAssetHolder(): AssetHolder

    fun addAssets()

    fun withNamespace(namespace: String?, cb: ()->Unit) {
        getAssetHolder().withNamespace(namespace, cb)
    }

    fun addJs(jsAsset: JsPage, configure: JsPage.()->Unit = {}): JsPage {
        return getAssetHolder().addJs(jsAsset).also(configure)
    }

    fun addJs(jsAsset: String, configure: JsPage.()->Unit = {}): JsPage {
        return getAssetHolder().addJs(jsAsset).also(configure)
    }

    fun addCss(cssAsset: CssPage, configure: CssPage.()->Unit = {}): CssPage {
        return getAssetHolder().addCss(cssAsset).also(configure)
    }

    fun addCss(cssAsset: String, configure: CssPage.()->Unit = {}): CssPage {
        return getAssetHolder().addCss(cssAsset).also(configure)
    }

    fun getScripts(): List<JsPage> {
        return getAssetHolder().getScripts()
    }

    fun getStyles(): List<CssPage> {
        return getAssetHolder().getStyles()
    }

    fun clearAssets() {
        getAssetHolder().clearAssets()
    }
}
